lib/Dialect/HW/HWTypes.cpp

```diff
@@ -413,6 +414,30 @@ void TypeAliasType::print(DialectAsmPrinter &p) const {
   p << getMnemonic() << "<" << getRef() << ", " << getInnerType() << ">";
 }
 
+Optional<TypedeclOp> TypeAliasType::getDecl(Operation *op) {
+  SymbolRefAttr ref = getRef();
```

Comment:
Shouldn't all these error conditions be checked in verification so that they don't need to be checked here?

Reply:
I would love to do that in the type verifier, but unfortunately that hook doesn't have access to the rest of the IR. It's a similar issue to why we need to have the inner type printed/parsed in the IR. The types are printed/parsed/verified in a vacuum where doing a symbol lookup doesn't appear to be an option.